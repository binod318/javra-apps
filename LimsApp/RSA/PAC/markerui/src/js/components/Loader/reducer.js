const loader = (state = 0, action) => {
  let source = "";
  if (action.func) source = action.func;
  else source = "undefine";

  switch (action.type) {
    case "LOADER_RESET":
      return 0;
    case "LOADER_SHOW":
      return state + 1;
    case "LOADER_HIDE":
      if (state > 0) {
        return state - 1;
      }
      break;
    default:
      return state;
  }
  return null;
};
export default loader;
