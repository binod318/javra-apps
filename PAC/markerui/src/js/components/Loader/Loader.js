import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import "./loader.scss";

export const myLoader = ({ status }) => {
  if (status <= 0) {
    return <span />;
  }

  return (
    <div className='loaderWrapper active' id='loader'>
      <div className='loader'>
        <i className='demo-icon icon-spin6 animate-spin' />
      </div>
    </div>
  );
};

const mapStateToProps = (state) => ({
  status: state.loader,
});

myLoader.defaultProps = {
  status: 0,
};
myLoader.propTypes = {
  status: PropTypes.number,
};
const Loader = connect(mapStateToProps, null)(myLoader);
export default Loader;
