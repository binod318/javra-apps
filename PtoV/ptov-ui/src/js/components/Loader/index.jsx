import React from 'react';
import Wrapper from '../Wrapper';

const Loader = () => (
  <Wrapper>
    <div className="loaderWrapper">
      <i className="icon icon-spin4 animate-spin" />
    </div>
  </Wrapper>
);
export default Loader;
