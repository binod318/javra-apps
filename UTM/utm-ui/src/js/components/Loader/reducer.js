const loader = (state = 0, action) => {
  switch (action.type) {
    case 'LOADER_RESET':
      return 0;
    case 'LOADER_SHOW':
      // concon(action, state, '++');
      return state + 1;
    case 'LOADER_HIDE':
      // concon(action, state, '--');
      if (state > 0) {
        return state - 1;
      }
      break;
    default:
      return state;
  }
  return null;
};
export default loader;
