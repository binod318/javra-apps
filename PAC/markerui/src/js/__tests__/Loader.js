import React from 'react';
import ReactDOM from 'react-dom';
// setup file
import { configure } from 'enzyme';
import { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import { LoaderW } from '../components/Loader/Loader';
configure({ adapter: new Adapter() });
// test file

describe('<LoaderW />', () => {
  const props = { status: 0 };
  const wrapper = shallow(<LoaderW {...props} />);

  it('LoaderW draw', () => {
    expect(wrapper.find('span').length).toBe(1);

    wrapper.setProps({ status: 1 });
    expect(wrapper.find('.loader').length).toBe(1);
  });

  it('match a loader shap', () => {
    const tree = shallow(<wrapper />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
