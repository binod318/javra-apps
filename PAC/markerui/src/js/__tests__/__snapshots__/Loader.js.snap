// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoaderW /> match a loader shap 1`] = `<wrapper />`;
